cmake/common/compiler_common.cmake

```diff
@@ -0,0 +1,89 @@
+# CMake common compiler options module
+
+include_guard(GLOBAL)
+
+# Set C and C++ language standards to C17 and C++17
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.17)
+  set(CMAKE_C_STANDARD 17)
+else()
+  set(CMAKE_C_STANDARD 11)
+endif()
+set(CMAKE_C_STANDARD_REQUIRED TRUE)
+set(CMAKE_C_EXTENSIONS FALSE)
```

Comment:
Didn't this cause issues on Linux/POSIX?

Reply:
I'm not aware of any. I've compiled OBS itself with that setting without issue, albeit only on Ubuntu.

Reply:
See:
* https://github.com/obsproject/obs-studio/blob/29.1.1/cmake/Modules/CompilerConfig.cmake#L17-L20
* https://github.com/obsproject/obs-studio/pull/8684

Reply:
Thanks, that allowed me to investigate the root causes for that:

It seems that on Linux `glibc` more or less relies on GNU extensions for a lot of core functionality. On obs-studio we merged code in `darray.h` that _requires_ them even, so disabling extensions on Linux will not be possible.

Thus I limited that setting to macOS and Windows.

Reply:
Ah, I see now.
* the darray test code in `darray.h` requires C extensions and is ifdef'd to only be included on Linux
* thus it fails on Linux, because we tried to globally disable C extensions and the code requires them

That is _unfortunate_.

Reply:
On closer look, it seems like that code should only be hit `if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)` and the builder/user/dev also left `ENABLE_DARRAY_TYPE_TEST` set to `ON` (which admittedly is default here).

Is this file used for settings to build OBS itself or just the plugin? If it's _just_ the plugin, then we could probably leave C extensions disabled across the board to discourage extension usage. Might we be able to check the value of `ENABLE_DARRAY_TYPE_TEST` to toggle it on and off? Otherwise, I suppose we can leave `set(CMAKE_C_EXTENSIONS FALSE)` limited to macOS and Windows.

Reply:
The header itself (as with all the other headers in libobs) is unfortunately always imported by libobs' headers (I'd argue we expose/force to include way too much stuff, but libobs' header infrastructure has never been designed with standalone usage in mind).

As such the code and lots of other stuff is automatically added to a plugin's codebase and requires the same flags.

Reply:
FWIW it seems `threading.h` in the utils directory also requires GNU extensions to be enabled on Linux, as some of the `pthread` functionality we use is only enabled in that header if `gcc` runs in a non-standards mode.